File: app/src/test/java/com/kieran/winnipegbustests/TripPlannerTest.java
package com.kieran.winnipegbustests;

import com.kieran.winnipegbusbackend.LoadResult;
import com.kieran.winnipegbusbackend.TransitApiManager;
import com.kieran.winnipegbusbackend.TripPlanner.LocationFactory;
import com.kieran.winnipegbusbackend.TripPlanner.classes.Location;
import com.kieran.winnipegbusbackend.TripPlanner.classes.Trip;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;
import org.junit.Assert;
import org.junit.Test;

public class TripPlannerTest {
    @Test
    public void CompleteTripParse() throws JSONException {
        final String url = "https://api.winnipegtransit.com/v2/trip-planner.json?origin=addresses/130038&destination=addresses/203323&api-key=FTy2QN8ts293ZlhYP1t&usage=short";
        LoadResult<JSONObject> result = TransitApiManager.getJson(url);
        JSONObject object = result.getResult();
        JSONArray plans = object.getJSONArray("plans");
        Trip[] trips = new Trip[plans.length()];

        for(int i = 0; i < plans.length(); i++) {
            trips[i] = new Trip(plans.getJSONObject(i));
        }

        Assert.assertTrue(trips.length > 0);
    }

    @Test
    public void GetAddressLocationFromQueryAndParse() throws JSONException {
        String url = TransitApiManager.generateLocationQueryUrl("1099 Grant");
        LoadResult<JSONObject> result = TransitApiManager.getJson(url);
        JSONArray locations = result.getResult().getJSONArray("locations");
        Location location = LocationFactory.createLocation(locations.getJSONObject(0));

        Assert.assertNotNull(location);
        Assert.assertTrue(location.getTitle().equals("1099 GrantAve"));
    }

    @Test
    public void GetIntersectionLocationFromQueryAndParse() throws JSONException {
        String url = TransitApiManager.generateLocationQueryUrl("Grant at Stafford");
        LoadResult<JSONObject> result = TransitApiManager.getJson(url);
        JSONArray locations = result.getResult().getJSONArray("locations");
        Location location = LocationFactory.createLocation(locations.getJSONObject(0));

        Assert.assertNotNull(location);
        Assert.assertTrue(location.getTitle().equals("GrantAve at StaffordSt"));
    }

    @Test
    public void GetMonumentLocationFromQueryAndParse() throws JSONException {
        String url = TransitApiManager.generateLocationQueryUrl("mts");
        LoadResult<JSONObject> result = TransitApiManager.getJson(url);
        JSONArray locations = result.getResult().getJSONArray("locations");
        Location location = LocationFactory.createLocation(locations.getJSONObject(0));

        Assert.assertNotNull(location);
        Assert.assertTrue(location.getTitle().equals("Bell MTS Place"));
    }
}

File: app/src/test/java/com/kieran/winnipegbustests/TripPlannerTest.kt
package com.kieran.winnipegbustests

import com.kieran.winnipegbusbackend.winnipegtransit.TransitApiManager
import com.kieran.winnipegbusbackend.TripPlanner.LocationFactory
import com.kieran.winnipegbusbackend.TripPlanner.classes.Trip

import org.json.JSONException
import org.junit.Assert
import org.junit.Test

class TripPlannerTest {

    @Test
    @Throws(JSONException::class)
    fun GetAddressLocationFromQueryAndParse() {
        val url = TransitApiManager.generateLocationQueryUrl("1099 Grant")
        val result = TransitApiManager.getJson(url)
        val locations = result.result!!.getJSONArray("locations")
        val location = LocationFactory.createLocation(locations.getJSONObject(0))

        Assert.assertNotNull(location)
        Assert.assertTrue(location.title == "1099 GrantAve")
    }

    @Test
    @Throws(JSONException::class)
    fun GetIntersectionLocationFromQueryAndParse() {
        val url = TransitApiManager.generateLocationQueryUrl("Grant at Stafford")
        val result = TransitApiManager.getJson(url)
        val locations = result.result!!.getJSONArray("locations")
        val location = LocationFactory.createLocation(locations.getJSONObject(0))

        Assert.assertNotNull(location)
        Assert.assertTrue(location.title == "GrantAve at StaffordSt")
    }

    @Test
    @Throws(JSONException::class)
    fun GetMonumentLocationFromQueryAndParse() {
        val url = TransitApiManager.generateLocationQueryUrl("mts")
        val result = TransitApiManager.getJson(url)
        val locations = result.result!!.getJSONArray("locations")
        val location = LocationFactory.createLocation(locations.getJSONObject(0))

        Assert.assertNotNull(location)
        Assert.assertTrue(location.title == "Bell MTS Place")
    }
}
